client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,64 @@ def build_template(
+                enabled we assume the user tries to load it only into a
+                "New File" (unsaved empty workfile).
+                When False, the default value, we assume we explicitly want to
+                build the template in our current scene regardless of current
+                scene state.
 
         """
-        if any(
-            value is None
-            for value in [
-                template_path,
-                keep_placeholders,
-                create_first_version,
-            ]
-        ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
-            if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
-            if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
 
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
-
-        # if first version is created, import template
-        # and populate placeholders
+        # Get default values if not provided
         if (
-            create_first_version
-            and workfile_creation_enabled
-            and created_version_workfile
+            template_path is None
+            or keep_placeholders is None
+            or create_first_version is None
         ):
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
+            if keep_placeholders is None:
+                keep_placeholders: bool = preset["keep_placeholder"]
+            if create_first_version is None:
+                create_first_version: bool = preset["create_first_version"]
+
+        # Build the template if we are explicitly requesting it or if it's
+        # an unsaved "new file".
+        is_new_file = not self.host.get_current_workfile()
+        explicit_build_requested = not workfile_creation_enabled
+        if is_new_file or explicit_build_requested:
+            self.log.info(f"Building the workfile template: {template_path}")
             self.import_template(template_path)
             self.populate_scene_placeholders(
                 level_limit, keep_placeholders)
 
-            # save workfile after template is populated
-            self.save_workfile(created_version_workfile)
+        if not workfile_creation_enabled:
+            # Do not consider saving a first workfile version, if this
+            # is not set to be a "workfile creation". Then we assume
+            # only the template should get build.
+            return
 
-        # ignore process if first workfile is enabled
-        # but a version is already created
-        if workfile_creation_enabled:
+        if not create_first_version:
+            # Do not save whatsoever
```

Comment:
this is not really that helpful to me, I'd probably add the docstring before saying something like:
`# Return early if create first version is disabled`
Or... I would change it to encapsulate the create version code inside the conditional:
```python
# Create the first workfile version if requested
if create_version_version:
        workfile_path = self.get_workfile_path()
        if not os.path.exists(workfile_path):
            self.log.info("Saving first workfile: %s", workfile_path)
            self.save_workfile(workfile_path)
        else:
            self.log.info(
                "A workfile already exists. Skipping save of workfile as "
                "initial version.")
```